charts/delphai-keda/templates/deployment.yml

```diff
@@ -71,3 +72,68 @@ spec:
               containerPort: {{ .Values.http.port }}
             - name: h2c
               containerPort: {{ .Values.grpc.port }}
+
+        {{ if and .Values.http.enabled .Values.http.public .Values.http.publicWithOAuth }}
+        - name: oauth2-proxy
+          image: "quay.io/oauth2-proxy/oauth2-proxy:v7.2.0"
+          args:
+            - '--email-domain=*'
+          env:
+          - name: OAUTH2_PROXY_HTTP_ADDRESS
+            value: 0.0.0.0:{{ .Values.http.oAuthProxyPort }}
+
+          - name: OAUTH2_PROXY_PROVIDER
+            value: "oidc"
+
+          - name: OAUTH2_PROXY_OIDC_ISSUER_URL
+            valueFrom:
+              secretKeyRef:
+                name: oauth2-proxy
```

Comment:
I've created this secret manually in all clusters, I'll create a ticket to generate it with TF